src/WebJobs.Extensions.DurableTask/AzureStorageDurabilityProvider.cs

```diff
@@ -242,16 +251,22 @@ public override bool TryGetScaleMonitor(
             string connectionName,
             out IScaleMonitor scaleMonitor)
         {
-            CloudStorageAccount storageAccount = this.storageAccountProvider.GetStorageAccountDetails(connectionName).ToCloudStorageAccount();
-            DurableTaskMetricsProvider metricsProvider = this.GetMetricsProvider(functionName, hubName, storageAccount, this.logger);
-            scaleMonitor = new DurableTaskScaleMonitor(
-                functionId,
-                functionName,
-                hubName,
-                storageAccount,
-                this.logger,
-                metricsProvider);
-            return true;
+            lock (this.initLock)
+            {
```

Comment:
a bit nervous about a lock in the ScaleController (and I'm not sure it is really needed, I suspect other instances of the scaler class will be garbage collected away so long as the use the same ID), but the implementation seems safe.

Reply:
I'm not sure if it's needed either, but any particular thing that makes you nervous? I'm fine with removing it.